crates/valence_protocol/src/packets/s2c/look_at.rs

```diff
@@ -0,0 +1,60 @@
+use std::io::Write;
+
+use crate::{Decode, DecodePacket, Encode, EncodePacket, VarInt};
+
+#[derive(Copy, Clone, PartialEq, Debug, EncodePacket, DecodePacket)]
+#[packet_id = 0x37]
+pub struct LookAt {
+    pub feet_eyes: FeetEyes,
+    pub target_position: [f64; 3],
+    pub entity_id: Option<VarInt>,
+    pub entity_feet_eyes: Option<FeetEyes>,
```

Comment:
Looks like you can bundle these two fields into an `Option<LookAtEntity>`. Then you wouldn't need a manual `Encode`/`Decode` impl and it would more accurately reflect the states this packet can be in.

```rust
struct LookAtEntity {
    pub entity_id: VarInt,
    pub entity_feet_eyes: FeetEyes
}
```